src/components/itemPage/Item.jsx

```diff
@@ -89,8 +90,41 @@ class Item extends Component {
     );
   }
 
+  onlistening = async () => {
```

Comment:
`onX` is usually the name of a function you give to a "handler" as a parameter.
This name is misleading since you call it directly (i.e. you trigger the action yourself, hence it should be phrased as an order).